dynawo/sources/Models/CPP/Events/ModelVoltageSetPointChange/DYNModelVoltageSetPointChange.cpp

```diff
@@ -0,0 +1,258 @@
+#include "DYNVariableForModel.h"
+#include "DYNParameter.h"
+
+using std::min;
+using std::vector;
+using std::string;
+
+using std::stringstream;
+using std::map;
+
+using boost::shared_ptr;
+
+using parameters::ParametersSet;
+
+/**
+ * @brief ModelVoltageSetPointChangeFactory getter
+ *
+ * @return A pointer to a new instance of ModelVariationaAreaFactory
+ */
+extern "C" DYN::SubModelFactory* getFactory() {
+  return (new DYN::ModelVoltageSetPointChangeFactory());
+}
+
+/**
+ * @brief ModelVoltageSetPointChangeFactory destroy method
+ */
+extern "C" void deleteFactory(DYN::SubModelFactory* factory) {
+  delete factory;
+}
+
+/**
+ * @brief ModelVoltageSetPointChange getter
+ *
+ * @return A pointer to a new instance of ModelVoltageSetPointChange
+ */
+extern "C" DYN::SubModel* DYN::ModelVoltageSetPointChangeFactory::create() const {
+  DYN::SubModel * model(new DYN::ModelVoltageSetPointChange());
+  return model;
+}
+
+/**
+ * @brief ModelVoltageSetPointChange destroy method
+ */
+extern "C" void DYN::ModelVoltageSetPointChangeFactory::destroy(DYN::SubModel* model) const {
+  delete model;
+}
+
+namespace DYN {
+
+ModelVoltageSetPointChange::ModelVoltageSetPointChange() :
+Impl("VoltageSetPointChange"),
+startTime_(0.),
+stopTime_(0.),
+numLoads_(0) {
+}
+
+void
+ModelVoltageSetPointChange::init(const double /*t0*/) {
+  // not needed
+}
+
+void
+ModelVoltageSetPointChange::initializeFromData(const boost::shared_ptr<DataInterface>& /*data*/) {
+  // not needed
+}
+
+void
+ModelVoltageSetPointChange::initializeStaticData() {
+  // not needed
+}
+
+void
+ModelVoltageSetPointChange::getSize() {
+  sizeF_ = 0;
+  sizeY_ = 0;
+  sizeZ_ = numLoads_;  // number of loads controlled
+  sizeG_ = 2;
+  sizeMode_ = 0;
+
+  calculatedVars_.assign(nbCalculatedVars_, 0);
+}
+
+void
+ModelVoltageSetPointChange::evalF(double /*t*/, propertyF_t /*type*/) {
+  // not needed
+}
+
+void
+ModelVoltageSetPointChange::evalG(const double t) {
+  gLocal_[0] = ((t > startTime_ || doubleEquals(t, startTime_)) && t < stopTime_) ? ROOT_UP : ROOT_DOWN;
+  gLocal_[1] = (t > stopTime_ || doubleEquals(t, stopTime_)) ? ROOT_UP : ROOT_DOWN;
+}
+
+void
+ModelVoltageSetPointChange::setFequations() {
+  // not needed
+}
+
+void
+ModelVoltageSetPointChange::setGequations() {
+  gEquationIndex_[0] = "stopTime > t >= startTime";
+  gEquationIndex_[1] = "t >= stopTime";
+
+  assert(gEquationIndex_.size() == (unsigned int) sizeG() && "Model VoltageSetPointChange: gEquationIndex.size() != gLocal_.size()");
+}
+
+void
+ModelVoltageSetPointChange::evalJt(const double /*t*/, const double /*cj*/, SparseMatrix& /*jt*/, const int /*rowOffset*/) {
+  // not needed
+}
+
+void
+ModelVoltageSetPointChange::evalJtPrim(const double /*t*/, const double /*cj*/, SparseMatrix& /*jt*/, const int /*rowOffset*/) {
+  // not needed
+}
+
+void
+ModelVoltageSetPointChange::evalZ(const double /*t*/) {
+  if (gLocal_[0] == ROOT_UP) {
+    for (int i = 0; i < numLoads_; ++i) {
+      zLocal_[i] = -0.05;
```

Comment:
I was expecting the decrease level to be a parameter of the model: it would be nicer in my opinion and more generic. 5% is perhaps a specific value

Reply:
It was a question I had so I will do it

Reply:
done